### PR TITLE
Project

### PR DESCRIPTION
**Pull request by fyunes**
project

### CODE REVIEW

File: project/wowstat/src/app/guild-search.service.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  constructor(private http: Http) { }
+
+  getGuild(guild: String, realm: String): Observable<GuildInterface> {
+    return this.http.get(`https://us.api.battle.net/wow/guild/${realm}/${guild}?fields=members&locale=en_US&apikey=4aq73rmdfp746j669j2u33qw434pbtj3`).map((data) => {
+      return data.json();
+    });
+  }
+
+}
+
```
Comment (DemianMoschin-GLB) [inline project/wowstat/src/app/guild-search.service.ts]: Remove empty lines

File: project/wowstat/src/app/guild-search/guild-search.component.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { GuildMembers } from '../guild-members';
+
+@Component({
+  selector: 'app-guild-search',
+  templateUrl: './guild-search.component.html',
+  styleUrls: ['./guild-search.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class GuildSearchComponent implements OnInit {
+
```
Comment (DemianMoschin-GLB) [inline project/wowstat/src/app/guild-search/guild-search.component.ts]: remove empty lines

File: project/wowstat/src/app/leader-board.service.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+@Injectable()
+export class LeaderBoardService {
+
+
+
+  constructor(private http: Http) { }
+
+  getAll(rank: String): Observable<vs[]> {
+    return this.http.get(`https://us.api.battle.net/wow/leaderboard/${rank}?locale=en_US&apikey=4aq73rmdfp746j669j2u33qw434pbtj3`)
```
Comment (DemianMoschin-GLB) [inline project/wowstat/src/app/leader-board.service.ts]: constants such as keys, could be put inside an external file, so you can share it across your api calls

File: project/wowstat/src/app/realm-status.service.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import 'rxjs/add/operator/map';
+import { RealmInterface } from '../app/realm-interface'
+
+@Injectable()
+export class RealmStatusService {
+
+  constructor(private http: Http) { }
+  
+    getAll(): Observable<RealmInterface[]> {
+      return this.http.get(`https://us.api.battle.net/wow/realm/status?locale=en_US&apikey=4aq73rmdfp746j669j2u33qw434pbtj3`)
```
Comment (DemianMoschin-GLB) [inline project/wowstat/src/app/realm-status.service.ts]: same as above

File: project/wowstat/src/app/realm-status/realm-status.component.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  ngOnInit() {
+  }
+
+  search() {
+    this.RealmStatusService.getAll().subscribe(arrayRealms => {
+      this.arrayRealms = arrayRealms;
+    })
+  }
+}
+
```
Comment (DemianMoschin-GLB) [inline project/wowstat/src/app/realm-status/realm-status.component.ts]: remove empty lines

File: project/wowstat/src/app/guild-search.service.ts (hunk 1 above)
Comment (fyunes) [reply project/wowstat/src/app/guild-search.service.ts]: done

File: project/wowstat/src/app/guild-search/guild-search.component.ts (hunk 2 above)
Comment (fyunes) [reply project/wowstat/src/app/guild-search/guild-search.component.ts]: done

File: project/wowstat/src/app/leader-board.service.ts (hunk 3 above)
Comment (fyunes) [reply project/wowstat/src/app/leader-board.service.ts]: done, also added a dropdown in character search for servers so theres no need to type it down

File: project/wowstat/src/app/realm-status.service.ts (hunk 4 above)
Comment (fyunes) [reply project/wowstat/src/app/realm-status.service.ts]: done

File: project/wowstat/src/app/realm-status/realm-status.component.ts (hunk 5 above)
Comment (fyunes) [reply project/wowstat/src/app/realm-status/realm-status.component.ts]: done

**Pull request merged**